code/game/atoms_movable.dm

```diff
@@ -627,16 +662,15 @@
 		var/area/old_area = get_area(oldloc)
 		var/area/destarea = get_area(destination)
 
-		loc = destination
+		update_loc(destination)
 		moving_diagonally = 0
 
 		if(!same_loc)
 			if(oldloc)
+				SEND_SIGNAL(oldloc, COMSIG_MOVABLE_UNCROSSED, src)
```

Comment:
I'm still uncomfortable with this signal being fired from multiple places--signals should only come from one place.

Reply:
now its only sent from atom/exited